expression/integration_test.go

```diff
@@ -7127,3 +7127,14 @@ func (s *testIntegrationSuite) TestIssue19504(c *C) {
 	tk.MustQuery("select (select count(c_int) from t2 where c_int = t1.c_int) c1, (select count(1) from t2 where c_int = t1.c_int) c2 from t1;").
 		Check(testkit.Rows("1 1", "0 0", "0 0"))
 }
+
+func (s *testIntegrationSuite) TestIssue19596(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("create table t (a int) partition by range(a) (PARTITION p0 VALUES LESS THAN (10));")
+	tk.MustGetErrMsg("alter table t add partition (partition p1 values less than (a));", "[ddl:1659]Field '`a`' is of a not allowed type for this type of partitioning")
+	tk.MustQuery("select * from t;")
+	tk.MustExec("drop table if exists t;")
+	tk.MustGetErrMsg("create table t (a int) partition by range(a) (PARTITION p0 VALUES LESS THAN (a));", "[ddl:1659]Field '`a`' is of a not allowed type for this type of partitioning")
```

Comment:
This error code/message seems not compatible with MySQL:
```sql
create table t (
  a int
) partition by range(a) (
  partition p0 values less than (1)
);

alter table t add partition (partition p1 values less than (a));
```

```console
(1054, "Unknown column 'a' in 'partition function'")
```

Reply:
it return error msg `[expression:1054]Unknown column 'a' in 'expression'` now, it is returned by `simpleRewriter`